bin/dump_wiki

```diff
@@ -38,9 +41,72 @@ if (argv.h) {
 // Strip a trailing slash
 argv.host = argv.host.replace(/\/$/, '');
 
-argv.ns = Number(argv.ns);
+// Default global options
+var options = { maxConcurrency : 50, concurrency: 3 };
+
+if (argv.c) {
+    // Get config, or throw exception on error
+    try {
+        var conf = yaml.safeLoad(fs.readFileSync(argv.c));
+        Object.keys(conf.dump_wiki).forEach(function(arg){
+            if (!argv[arg]) {
+                argv[arg] = conf.dump_wiki[arg];
+            }
+        });
+        options = conf.global_opts;
+    } catch (e) {
+      console.log(e);
+    }
+
+    // Check if required params are missing
+    if (!argv.apiURL || !argv.domain || !argv.ns || !argv.host) {
+        console.log('[Error - One of the required params is missing in config]');
+        process.exit(1);
+    }
+
+    // Set defaults
+    if (!argv.d) {
+        argv.d = '';
+    }
+    if (!argv.db) {
+        argv.db = '';
+    }
+    if (!argv.verbose) {
+        argv.verbose = true;
+    }
+
+} else {
+    //check whats missing and show help
+    argParser.reset()
+        .usage('Create a HTML dump in a subdir\n'
+                + '\nExample:\n$0 --domain en.wikipedia.org --ns 0 --apiURL http://en.wikipedia.org/w/api.php')
+        .demand(['apiURL', 'domain', 'ns', 'host'])
+        .options('h', {
+            alias: 'help'
+        })
+        .alias('domain', 'prefix')
+        .options('d', {
+            alias : 'saveDir',
+            default : ''
+        })
+        .options('db', {
+            alias : 'dataBase',
+            default : ''
+        })
+        .options('verbose', {
+            default : true
+        })
+        //.default('apiURL', 'http://en.wikipedia.org/w/api.php')
+        //.default('prefix', 'en.wikipedia.org')
+        //.default('ns', '0')
+        .default('host', 'http://rest.wikimedia.org');
+    argv = argParser.argv;
+    if (!argv) {
+        console.log(argv);
+    }
+}
 
-return makeDump(argv)
+return makeDump(argv, options)
```

Comment:
I passing both of them separately because the makeDump function will separates them again and  use them differently also 'argv' contains arguments and options contain global options for the script.
Do you still want me to merge in into one object?